### PR TITLE
move f.Command out of the factory

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
NONE
```

Moves the "f.Command" method out of factory_client_access

cc @soltysh

### CODE REVIEW

File: pkg/kubectl/polymorphichelpers/helpers.go [hunk 1]
```diff
@@ -233,3 +238,36 @@ func SelectorsForObject(object runtime.Object) (namespace string, selector label
 
 	return namespace, selector, nil
 }
+
+// StringifyCommand will stringify and return all environment arguments ie. a command run by a client
+// using the factory.
+// Set showSecrets false to filter out stuff like secrets.
+func StringifyCommand(cmd *cobra.Command, showSecrets bool) string {
```
Comment (deads2k) [inline pkg/kubectl/polymorphichelpers/helpers.go]: This should be in the same spot as the `RecordFlags`, right?

**Comment by deads2k** [comment]
Good intent, wrong package.

/approve
/hold

**Comment by juanvallejo** [comment]
@deads2k updated "StringifyCommand" to be a helper within record_flags

File: pkg/kubectl/genericclioptions/record_flags.go [hunk 2]
```diff
@@ -150,3 +155,36 @@ func (r *ChangeCauseRecorder) MakeRecordMergePatch(obj runtime.Object) ([]byte,
 
 	return jsonpatch.CreateMergePatch(oldData, newData)
 }
+
+// stringifyCommand will stringify and return all environment arguments ie. a command run by a client
+// using the factory.
+// Set showSecrets false to filter out stuff like secrets.
+func stringifyCommand(cmd *cobra.Command, showSecrets bool) string {
```
Comment (deads2k) [inline pkg/kubectl/genericclioptions/record_flags.go]: do you ever pass true?  If not, clean it up

File: pkg/kubectl/genericclioptions/record_flags.go [hunk 3]
```diff
@@ -60,12 +65,12 @@ func (f *RecordFlags) ToRecorder() (Recorder, error) {
 }
 
 // Complete is called before the command is run, but after it is invoked to finish the state of the struct before use.
-func (f *RecordFlags) Complete(changeCause string) error {
+func (f *RecordFlags) Complete(cmd *cobra.Command) error {
```
Comment (deads2k) [inline pkg/kubectl/genericclioptions/record_flags.go]: When you rebase, create a second function that still takes the string directly.

File: pkg/kubectl/genericclioptions/record_flags.go [hunk 4]
```diff
@@ -150,3 +155,36 @@ func (r *ChangeCauseRecorder) MakeRecordMergePatch(obj runtime.Object) ([]byte,
 
 	return jsonpatch.CreateMergePatch(oldData, newData)
 }
+
+// stringifyCommand will stringify and return all environment arguments ie. a command run by a client
+// using the factory.
+// Set showSecrets false to filter out stuff like secrets.
+func stringifyCommand(cmd *cobra.Command) string {
```
Comment (soltysh) [inline pkg/kubectl/genericclioptions/record_flags.go]: We need a better name, please. `stringifyCommand` completely hides the functionality of this command.

Comment (soltysh) [reply pkg/kubectl/genericclioptions/record_flags.go]: `classifyCommandOutput`, `hideCommandSecrets` or something like that.

Comment (soltysh) [reply pkg/kubectl/genericclioptions/record_flags.go]: And I agree with David, move this to one of the general helpers and make it public.

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/kubectl/genericclioptions/record_flags.go (hunk 4 above)
Comment (soltysh) [reply pkg/kubectl/genericclioptions/record_flags.go]: Ignore what I said above, I thought about something else ;)

**Comment by juanvallejo** [comment]
@deads2k @soltysh thanks, comments addressed

**Pull request merged**